### PR TITLE
Feature/associations

### PR DESCRIPTION
**Pull request by Tchilo**
- Add classroom file 
- Add book file 
- Add rental file
- Update person file

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :lable
```
Comment (Ghiftee) [inline classroom.rb]: - [x] Kindly correct the spelling of label here

File: classroom.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(lable)
+    @lable = lable
```
Comment (Ghiftee) [inline classroom.rb]: - [x] Kindly correct the spelling of label here.

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Classroom
```
Comment (Ghiftee) [inline classroom.rb:1]: - [x] You would need a getter method also for students in the Classroom class. ``` When adding a student to a classroom it also sets the classroom for the student.```

File: person.rb [hunk 4]
```diff
@@ -1,18 +1,30 @@
...
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @corrector = Corrector.new
+    @rentals = []
+  end
+
+  attr_accessor :name, :age
```
Comment (Ghiftee) [inline person.rb]: - [x] You would need to add getter and setter methods for rentals too.

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+require 'date'
+
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
```
Comment (Ghiftee) [inline rental.rb]: - [x] Kindly add getter and setter methods for person and book.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello 👋🏽  @Tchilo 

Good job working on the project so far! 
✔️ There are no linter errors
✔️ You have all the required classes
However, there are some issues that you need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: rental.rb (hunk 5 above)
Comment (Tchilo) [reply rental.rb]: Many thanks I will implement all the changes.

File: classroom.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :label
+  attr_reader : student
```
Comment (mikethreels) [inline classroom.rb]: - [ ] great job setting up the attribute accessor for student 👍  however when implementing the requested changes by the previous CR you made a very small mistake which will cause it not to work, please remove the space between the full colon and the word 'student'

File: person.rb [hunk 7]
```diff
@@ -1,18 +1,31 @@
...
+  attr_reader :id
+  attr_accessor :rentals
```
Comment (mikethreels) [inline person.rb]: - [ ] [OPTIONAL] you can add the  `attr_accesor` for `:rentals`  in the same line as `:name` and `:age`

File: rental.rb [hunk 8]
```diff
@@ -0,0 +1,16 @@
+require 'date'
+
+class Rental
+  attr_accessor :date
```
Comment (mikethreels) [inline rental.rb]: - [ ] [OPTIONAL] you can add the `attr_accessor` for `:data` in the same line as `:person` and `:book`

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi TC, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 8 above)
Comment (Tchilo) [reply rental.rb]: Sorry, I asked for a code review without fixing the errors. I have already made these changes.
Thanks for the review 😃

**Review by hiromataba — APPROVED** [review]
## STATUS: APPROVED :tada: :tada: :tada: 
Hi @Tchilo   :wave: , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**